### ISSUE TITLE
Ubuntu 20.04 .NET Core install failure: File has unexpected size: https://packages.microsoft.com/repos/microsoft-ubuntu-focal-prod/dists/focal

### ISSUE THREAD

**Issue by sindbach** [dotnet/core#5372]
# Issue Title

It appears the checksums in the InRelease file have not been updated after a release.
Related to issue [#4167](https://github.com/dotnet/core/issues/4167)

# General

Happens for dotnet-sdk-3.1 and dotnet-sdk-2.2 on ubuntu 18.04 and ubuntu 20.04 

Info for Ubuntu 20.04 / dotnet-sdk-3.1
```
Step 7/19 : RUN apt-get update && apt-get install -y     dotnet-sdk-3.1 &&     apt-get clean &&     rm -rf /var/lib/apt/lists/*
 ---> Running in 155b44eca864
Get:1 https://packages.microsoft.com/repos/microsoft-ubuntu-focal-prod focal InRelease [10.5 kB]
Get:2 http://security.ubuntu.com/ubuntu focal-security InRelease [107 kB]
Hit:3 http://archive.ubuntu.com/ubuntu focal InRelease
Get:4 https://packages.microsoft.com/repos/microsoft-ubuntu-focal-prod focal/main amd64 Packages [34.1 kB]
Err:4 https://packages.microsoft.com/repos/microsoft-ubuntu-focal-prod focal/main amd64 Packages
  File has unexpected size (34272 != 34103). Mirror sync in progress? [IP: 20.188.102.6 443]
  Hashes of expected file:
   - Filesize:34103 [weak]
   - SHA512:03535224e21b7d764e4a5683b314e5c7ade42cc1fb867d0971428324b08548c9338c0e014656eb6f994f1cbbbd9b21fa3be17fcb4c48ac53e9f9d36e81e966a5
   - SHA256:af87c84a025cd3a4777cf184d316f8cfa358685fd323475b565406d11b6fac4a
   - SHA1:1487c68359752eb6c9bbeeb158b34d7597e391df [weak]
   - MD5Sum:ef3b0918fca2da4c0b3213db7cd12984 [weak]
  Release file created at: Tue, 13 Oct 2020 22:37:20 +0000
Hit:5 http://archive.ubuntu.com/ubuntu focal-updates InRelease
Hit:6 http://archive.ubuntu.com/ubuntu focal-backports InRelease
Fetched 117 kB in 2s (78.0 kB/s)
Reading package lists...
E: Failed to fetch https://packages.microsoft.com/repos/microsoft-ubuntu-focal-prod/dists/focal/main/binary-amd64/Packages.gz  File has unexpected size (34272 != 34103). Mirror sync in progress? [IP: 20.188.102.6 443]
   Hashes of expected file:
    - Filesize:34103 [weak]
    - SHA512:03535224e21b7d764e4a5683b314e5c7ade42cc1fb867d0971428324b08548c9338c0e014656eb6f994f1cbbbd9b21fa3be17fcb4c48ac53e9f9d36e81e966a5
    - SHA256:af87c84a025cd3a4777cf184d316f8cfa358685fd323475b565406d11b6fac4a
    - SHA1:1487c68359752eb6c9bbeeb158b34d7597e391df [weak]
    - MD5Sum:ef3b0918fca2da4c0b3213db7cd12984 [weak]
   Release file created at: Tue, 13 Oct 2020 22:37:20 +0000
E: Some index files failed to download. They have been ignored, or old ones used instead.
```

Info for Ubuntu 18.04 / dotnet-sdk-2.2

```
Step 6/19 : RUN echo "deb [arch=amd64] https://packages.microsoft.com/repos/microsoft-ubuntu-bionic-prod bionic main" > /etc/apt/sources.list.d/dotnetdev.list
 ---> Running in 4c761ceaccf5
Removing intermediate container 4c761ceaccf5
 ---> 7aeee6e27afa
Step 7/19 : RUN apt-get update && apt-get install -y     dotnet-sdk-2.2 &&     apt-get clean &&     rm -rf /var/lib/apt/lists/*
 ---> Running in 926c8daf5e51
Get:1 https://packages.microsoft.com/repos/microsoft-ubuntu-bionic-prod bionic InRelease [4003 B]
Hit:2 http://security.ubuntu.com/ubuntu bionic-security InRelease
Get:3 https://packages.microsoft.com/repos/microsoft-ubuntu-bionic-prod bionic/main amd64 Packages [156 kB]
Err:3 https://packages.microsoft.com/repos/microsoft-ubuntu-bionic-prod bionic/main amd64 Packages
  File has unexpected size (156011 != 155826). Mirror sync in progress? [IP: 20.188.102.6 443]
  Hashes of expected file:
   - Filesize:155826 [weak]
   - SHA512:89de923d7c9395f6fd289afc05795fc29318b668727653ab024e33ce85a5bf909f59ce52efc9ca0cdfe1cc5ce7451796f1b38964d046913658ba86f9ffd97662
   - SHA256:18e2eff37d95013f29f7c68baf8f71954346a7118bc79de8246023cd78f8d853
   - SHA1:aab0617c5532798ffa6caf014f2e6b51035625b9 [weak]
   - MD5Sum:a3c30ff695b02ebe310c0852563b288a [weak]
  Release file created at: Tue, 13 Oct 2020 22:35:00 +0000
Hit:4 http://archive.ubuntu.com/ubuntu bionic InRelease
Hit:5 http://archive.ubuntu.com/ubuntu bionic-updates InRelease
Hit:6 http://archive.ubuntu.com/ubuntu bionic-backports InRelease
Fetched 4003 B in 2s (2402 B/s)
Reading package lists...
E: Failed to fetch https://packages.microsoft.com/repos/microsoft-ubuntu-bionic-prod/dists/bionic/main/binary-amd64/Packages.gz  File has unexpected size (156011 != 155826). Mirror sync in progress? [IP: 20.188.102.6 443]
   Hashes of expected file:
    - Filesize:155826 [weak]
    - SHA512:89de923d7c9395f6fd289afc05795fc29318b668727653ab024e33ce85a5bf909f59ce52efc9ca0cdfe1cc5ce7451796f1b38964d046913658ba86f9ffd97662
    - SHA256:18e2eff37d95013f29f7c68baf8f71954346a7118bc79de8246023cd78f8d853
    - SHA1:aab0617c5532798ffa6caf014f2e6b51035625b9 [weak]
    - MD5Sum:a3c30ff695b02ebe310c0852563b288a [weak]
   Release file created at: Tue, 13 Oct 2020 22:35:00 +0000
E: Some index files failed to download. They have been ignored, or old ones used instead.
```

**Comment by dagood** [comment]
This is probably related to the repository outage on Ubuntu 18.04: https://github.com/dotnet/core/issues/5373.

I checked the 20.04 feed at https://docs.microsoft.com/en-us/dotnet/core/install/linux-ubuntu#2004- (not quite the one you have, though?) and it's also getting the error:

```
Get:6 https://packages.microsoft.com/ubuntu/20.04/prod focal/main amd64 Packages [34.1 kB]
Err:6 https://packages.microsoft.com/ubuntu/20.04/prod focal/main amd64 Packages
  File has unexpected size (34272 != 34103). Mirror sync in progress? [IP: 13.90.21.104 443]
  Hashes of expected file:
   - Filesize:34103 [weak]
   - SHA512:03535224e21b7d764e4a5683b314e5c7ade42cc1fb867d0971428324b08548c9338c0e014656eb6f994f1cbbbd9b21fa3be17fcb4c48ac53e9f9d36e81e966a5
   - SHA256:af87c84a025cd3a4777cf184d316f8cfa358685fd323475b565406d11b6fac4a
   - SHA1:1487c68359752eb6c9bbeeb158b34d7597e391df [weak]
   - MD5Sum:ef3b0918fca2da4c0b3213db7cd12984 [weak]
  Release file created at: Tue, 13 Oct 2020 22:37:20 +0000
Fetched 10.5 kB in 1s (16.6 kB/s)
Reading package lists... Done
E: Failed to fetch https://packages.microsoft.com/ubuntu/20.04/prod/dists/focal/main/binary-amd64/Packages.gz  File has unexpected size (34272 != 34103). Mirror sync in progress? [IP: 13.90.21.104 443]
   Hashes of expected file:
    - Filesize:34103 [weak]
    - SHA512:03535224e21b7d764e4a5683b314e5c7ade42cc1fb867d0971428324b08548c9338c0e014656eb6f994f1cbbbd9b21fa3be17fcb4c48ac53e9f9d36e81e966a5
    - SHA256:af87c84a025cd3a4777cf184d316f8cfa358685fd323475b565406d11b6fac4a
    - SHA1:1487c68359752eb6c9bbeeb158b34d7597e391df [weak]
    - MD5Sum:ef3b0918fca2da4c0b3213db7cd12984 [weak]
   Release file created at: Tue, 13 Oct 2020 22:37:20 +0000
E: Some index files failed to download. They have been ignored, or old ones used instead.
```

@rbhanda

**Comment by sindbach** [comment]
Thanks @dagood 
I'm just following your comment on #4167 to open an issue when encountering this issue. Hopefully these issues show whomever prioritises the work to solve this. Just reading the history of the issue alone I could feel the pain for the .NET core team. 

This is been happening too long (~8 months?) for such a crucial part with no work around.

**Comment by dagood** [comment]
Absolutely, thanks for filing. I actually didn't realize this outage was happening on 20.04 too before you posted this. (Slightly edited the title to add that. We might want to merge the issues later on to track all of the current outage on https://github.com/dotnet/core/issues/5373.)

**Comment by sindbach** [comment]
You're right the reference of `focal` isn't too obvious to indicate 20.04. I should have included the 20.04 for clarity on the title. Thanks for the edit.

**Comment by dagood** [comment]
This is working for me now, the fix for 18.04 (https://github.com/dotnet/core/issues/5373#issuecomment-709827172) seems to have worked for 20.04 as well. Thanks @rbhanda for driving comms with the packages.microsoft.com team!

@sindbach can you confirm it's working for you too?

**Comment by sindbach** [comment]
Works now for Ubuntu 20.04 dotnet-sdk-3.1 , thanks @dagood !
